astropy/units/tests/test_physical.py

```diff
@@ -1,60 +1,457 @@
+    assert repr(pressure) == expected
+
+
+def test_physical_type_attribute_of_unit():
+    """
+    Test that a `u.Unit` instance has an attribute named `physical_type`
+    which is an instance of `physical.PhysicalType`.
+    """
+    assert isinstance(u.m.physical_type, physical.PhysicalType)
+
+
+def test_unknown_unit_physical_type_is_PhysicalType():
+    """
+    Test that a set of units with an unknown physical type has an
+    attribute named `physical_type` that is a `physical.PhysicalType`
+    instance.
+    """
+    unknown_unit = u.s ** 19
+    assert isinstance(unknown_unit.physical_type, physical.PhysicalType)
+
+
+def test_attempt_to_define_unknown_physical_type():
+    """Test that a unit cannot be defined as unknown."""
+    with pytest.raises(ValueError):
+        physical.def_physical_type(u.m ** 99, "unknown")
+
+
+@pytest.mark.parametrize("not_unit", [u.m.physical_type, 1, "1"])
+def test_get_physical_type_error(not_unit):
+    """
+    Test that attempting to get a physical type with
+    `physical.get_physical_type` for something that is not a unit raises
+    the appropriate exception.
+    """
+    with pytest.raises(TypeError):
+        physical.get_physical_type(not_unit)
+
+
+@pytest.mark.parametrize(
+    "temperature_unit1, temperature_unit2",
+    [
+        (u.K, u.deg_C),
+        (u.K, u.imperial.deg_F),
+        (u.deg_C, u.imperial.deg_F),
+        (u.K, u.imperial.deg_R),
+        (u.imperial.deg_F, u.imperial.deg_R),
+    ],
+)
+def test_physical_type_temperature_units(temperature_unit1, temperature_unit2):
+    """
+    Because K, °C, & °F have different physical type IDs, test that
+    different measurements of temperature are treated as equivalent.
+    """
+    assert temperature_unit1.physical_type == "temperature"
+    assert temperature_unit2.physical_type == "temperature"
+    assert temperature_unit1.physical_type == temperature_unit2.physical_type
+
+
+def test_physical_type_underscore_replacement():
+    """
+    Test that underscores are treated as spaces in a string
+    representation of a physical type.
+    """
+    pressure = u.Pa.physical_type
+    assert pressure == "energy_density"
+
+
+momentum_representations = [
+    "momentum",
+    "impulse",
+    "momentum/impulse",
+    "impulse/momentum",
+]
+
+
+@pytest.mark.parametrize("expected", momentum_representations)
+def test_momentum_equality_with_delimiters(expected):
+    """
+    Test that PhysicalType.__eq__ correctly treats different ways of
+    representing momentum and impulse.
+    """
+    momentum = (u.kg * u.m / u.s).physical_type
+    assert momentum == expected
+
+
+def test_physical_type_iteration():
+    """Test iterating through different physical type names."""
+    pressure = u.Pa.physical_type
+    physical_type_names = [physical_type_name for physical_type_name in pressure]
+    assert physical_type_names == ["energy density", "pressure", "stress"]
+
+
+def test_expanding_names_for_physical_type():
+    """
```

Comment:
The docstring is wrong here.  Also, you need to be sure to remove the definition from the physical types dict, i.e., some `try/finally` construct or `setup/teardown` (it may be an idea to move some functions in a test class with `setup_class` and `teardown_class` methods).